testsuite/testsuite.py

```diff
@@ -0,0 +1,36 @@
+#! /usr/bin/env python
+
+"""
+Usage::
+
+    testsuite.py [OPTIONS]
+
+Run the adareducer testsuite. This requires e3-testsuite.
+"""
+
+import os
+
+import e3.testsuite
+
+from drivers.shell_script import ShellScriptDriver
+
+
+class AdaReducerTestsuite(e3.testsuite.Testsuite):
+    tests_subdir = "tests"
+    test_driver_map = {"shell_script": ShellScriptDriver}
+    default_driver = "shell_script"
+
+    def set_up(self) -> None:
+        super().set_up()
+
+        # Set PYTHONPATH to find adareducer
+        root_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
+        os.environ["PYTHONPATH"] = "{}:{}".format(
+            root_dir,
+            os.environ.get("PYTHONPATH", ""),
+        )
+
+        os.environ["ADAREDUCER"] = f"python {root_dir}/ada_reduce.py"
```

Comment:
Is this really the way we go? Why can't we just add the dir to PATH? I've got traumatized of such environment variable abuses :)

Reply:
My only argument in favour of this is that it'll be convenient when we introduce coverage testing, we'll be able to just change this one line to a more complex command.
 
I don't think it'll pose a danger but perhaps I haven't been traumatized enough :-)

I don't have a strong opinion, feel free to open a PR to change this if you want.